client/main.py

```diff
@@ -1,11 +1,31 @@
 import os
+from typing import Text
 os.environ['PYGAME_HIDE_SUPPORT_PROMPT'] = "hide"
 import zmq
 import uuid
+import pygame as pg
 
 from definitions import *
 from gameSession import GameSession
 from connection import Connection
+from login import *
+from buttonPress import *
+
+pg.init()
```

Comment:
the logic for login shouldn't be here! you login.py contains code for a UI component Textbox which should be in its own file Textbox.py, then login.py needs to contain the mainloop for the login screen.

Reply:
so yes we need to do some cleanup here so that we can pass the game window around between the login screen and game session. Need to modify the api for run_session to take in a game_win instead